devnet-sdk/descriptors/chain_config.go

```diff
@@ -0,0 +1,183 @@
+package descriptors
+
+import (
+	"math/big"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+type ChainConfig struct {
+	InteropTime  *uint64                `json:"interopTime,omitempty"` // Interop switch time (nil = no fork, 0 = already on optimism interop)
+	Optimism     *params.OptimismConfig `json:"optimism,omitempty"`
+	BedrockBlock *big.Int               `json:"bedrockBlock,omitempty"` // Bedrock switch block (nil = no fork, 0 = already on optimism bedrock)
+	RegolithTime *uint64                `json:"regolithTime,omitempty"` // Regolith switch time (nil = no fork, 0 = already on optimism regolith)
+	CanyonTime   *uint64                `json:"canyonTime,omitempty"`   // Canyon switch time (nil = no fork, 0 = already on optimism canyon)
+	// Delta: the*Delta upgrade does not affect the execution-layer, and is thus not configurable in the chain config.
+	EcotoneTime  *uint64 `json:"ecotoneTime,omitempty"`  // Ecotone switch time (nil = no fork, 0 = already on optimism ecotone)
+	FjordTime    *uint64 `json:"fjordTime,omitempty"`    // Fjord switch time (nil = no fork, 0 = already on Optimism Fjord)
+	GraniteTime  *uint64 `json:"graniteTime,omitempty"`  // Granite switch time (nil = no fork, 0 = already on Optimism Granite)
+	HoloceneTime *uint64 `json:"holoceneTime,omitempty"` // Holocene switch time (nil = no fork, 0 = already on Optimism Holocene)
+	IsthmusTime  *uint64 `json:"isthmusTime,omitempty"`  // Isthmus switch time (nil = no fork, 0 = already on Optimism Isthmus)
+}
+
+// IsForkActivated checks if a specific fork is activated at the given timestamp
+// based on the chain configuration.
+func (c *ChainConfig) IsForkActivated(forkName rollup.ForkName, timestamp uint64) bool {
```

Comment:
@teddyknox I think we can reuse op-node's ChainSpec implementation for this. https://github.com/ethereum-optimism/optimism/blob/a33bba2fb32f0ddb350e9b672ac6a7207b607b5e/op-node/rollup/chain_spec.go#L166

Reply:
Will leave this pending the result of the discussion concerning which chain configuration we consider canonical.

Reply:
Switching back to op-geth's params.ChainConfig object